@here/harp-mapview/lib/MapView.ts

```diff
@@ -2438,13 +2464,14 @@ export class MapView extends THREE.EventDispatcher {
 
         const cameraPitch = MapViewUtils.extractAttitude(this, this.m_camera).pitch;
         const cameraPosZ = this.getCameraHeightAboveTerrain(TERRAIN_ZOOM_LEVEL);
+        const zoomLevelDistance = cameraPosZ / Math.cos(Math.min(cameraPitch, Math.PI / 3));
+        this.m_zoomLevel = MapViewUtils.calculateZoomLevelFromDistance(zoomLevelDistance, this);
+        this.m_fog.update(this, this.m_viewRanges.maximum);
 
         const target = MapViewUtils.rayCastWorldCoordinates(this, 0, 0);
         if (target !== null) {
-            this.m_lookAtDistance = target.sub(this.camera.position).length();
-            const zoomLevelDistance = cameraPosZ / Math.cos(Math.min(cameraPitch, Math.PI / 3));
-            this.m_zoomLevel = MapViewUtils.calculateZoomLevelFromDistance(zoomLevelDistance, this);
-            this.m_fog.update(this, this.m_viewRanges.maximum);
+            this.m_lookAtPoint.copy(target);
```

Comment:
we should have a better fallback instead of taking the last frames target. This will lead to non-deterministic behavior.

Reply:
This allows to keep the same pivot point for camera rotations. Imagine that we have focused on some point on globe (let's say north pole), when we are rotating camera around this point (pivot point) we could do it until ray is reaching the pole ground (earth surface) after cam angle (tilt) exceeds the 90 degrees ray hits the outer space, so there is not reference pivot point to rotate around. The proposed behavior would keep that last pivot and keep rotation around it.... Of course maybe there is something better then this... Here I am just trying to keep old controls implementation without re-writing entire code.
Any suggestions appreciated.

Reply:
I tried this locally:
```suggestion
        let target = null;
        if (this.projection.type === ProjectionType.Spherical) {
            target = MapViewUtils.rayCastWorldCoordinates(this, 0, 0);
        }

        if (target !== null) {
            this.m_lookAtDistance = target.sub(this.camera.position).length();
        } else {
            const cameraPosZ = this.getCameraHeightAboveTerrain(TERRAIN_ZOOM_LEVEL);
            const cameraPitch = MapViewUtils.extractAttitude(this, this.m_camera).pitch;
            const limitedPitch = Math.min(cameraPitch, 1.553343034); //Limit pitch to 89°
            this.m_lookAtDistance = cameraPosZ / Math.cos(limitedPitch);
        }

        this.m_zoomLevel = MapViewUtils.calculateZoomLevelFromDistance(this.m_lookAtDistance, this);
        this.m_fog.update(this, this.m_viewRanges.maximum);
```
So you would always rely on raycasting for globe but never for flat. Not sure how it will behave when exiting or entering the raycastable surface.

Reply:
So I have similar idea, with that difference that when rendering globe, our `limitedPitch` would be the the pitch to sphere tangent, so in other words, look at distance would be the distance to the furthest point on horizon (tangent point).
Although I would provide those changes progressively, splitting improvements into 2 commits, first with just raycasting simplification - the current one and the second that introduces new calculus. This way it will we easier to revert in case of regression.